unified-installer/astra-unified-installer.sh

```diff
@@ -2041,6 +2137,27 @@ EOF
     logdebug "$kustomization_file: added namespace transformer ($transformer_file_name)"
 }
 
+step_kustomize_global_pull_secret_if_needed() {
+    local -r global_pull_secret="${1:-""}"
+    local -r kustomization_file="${2}"
+    local -r kustomization_dir="$(dirname "$kustomization_file")"
+    [ -z "$global_pull_secret" ] && return 0
+
+    [ -z "$kustomization_file" ] && fatal "no kustomization file given"
+    [ ! -f "$kustomization_file" ] && fatal "kustomization file '$kustomization_file' does not exist"
+
+    insert_into_file_after_pattern "$kustomization_file" "patches:" '
+- target:
+    kind: Deployment
+  patch: |-
+    - op: replace
+      path: /spec/template/spec/imagePullSecrets
+      value:
+        - name: "'"${global_pull_secret}"'"
+'
+    logdebug "$kustomization_file: added pull secret patch ($global_pull_secret)"
+}
```

Comment:
Although we were generating the secrets as part of the kustomization file, we weren't actually setting the pull secrets for the Connector/Trident operators. This rectifies that.